proto/redpanda/api/dataplane/v1alpha1/pipeline.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Pipeline {
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.len = 20
```

Comment:
Where is the id coming from that it's required to be 20chars I wonder? I thought it could be an arbitrary id specified by the user. For example `my-snowflake-sink-connector`

Reply:
The ID should be generated by the backend that implements the service, that's why the create method doesn't receive the ID.
I'll clarify in a comment.

Reply:
Okay that makes sense to me. Maybe worth to add what kind of id it is supposed to be. I assume you had something specific in mind since you restricted the length to 20 chars.